### PR TITLE
Feature/cone

### PR DESCRIPTION
**Pull request by SBK78MC**

### CODE REVIEW

File: backend/RayTracing/Classes/Models/Object3D.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ def __init__(self, x=0, y=0, z=0, color=Color(), specular=100, reflection=0.1, t
         self.center = Vector(x, y, z)
         self.color = color
         self.specular = specular
-        self.reflection = reflection
+        self.reflection = self.reflectionCheck(reflection)
```
Comment (AlexDr7) [inline backend/RayTracing/Classes/Models/Object3D.py]: This is an unnecessary change, reflectionCheck is not used anymore. There is no reflectionCheck implemented.

File: backend/RayTracing/Classes/Models/Ray.py [hunk 2]
```diff
@@ -3,9 +3,21 @@
 
 class Ray:
 
-    def __init__(self, startp=Vector(0, 0, 0), direction=Vector(0, 0, 0)):
+    def __init__(self, startp=Vector(0, 0, 0), directionPoint=Vector(0, 0, 0)):
         self.startPoint = startp
-        self.direction = direction
+        self.direction = directionPoint.sub(startp)
```
Comment (AlexDr7) [inline backend/RayTracing/Classes/Models/Ray.py]: This change created a lot of bugs on the way current Rays work.
I tested it a bit using the old Rays and the new Rays and it creates a bug on reflection and I am guessing refraction as well. 
The reflectionRay direction and refractionRay direction are calculated in the RayTracer, so of course with the new Rays it does not work.

If we want to keep this change, I am guessing we should also have a function to normalize the Direction of the Ray from the RayTracer.

Comment (SBK78MC) [reply backend/RayTracing/Classes/Models/Ray.py]: Thank you for the explanation and the advice! Very good point. For the arbitrary camera position we need this calculation. Maybe we should change the RayTracer.

File: backend/RayTracing/Classes/Models/Object3D.py (hunk 1 above)
Comment (SBK78MC) [reply backend/RayTracing/Classes/Models/Object3D.py]: Fixed :)
